TestChaos is flaky

The following test appears to have failed:

[#10187](https://circleci.com/gh/cockroachdb/cockroach/10187):

```
I1215 02:44:53.894710 17886 client/db.go:478  failed batch: failed to send RPC request Server.Batch: client is unhealthy
I1215 02:44:57.358427 17886 acceptance/chaos_test.go:98  client 2 shutting down
I1215 02:44:57.426019 17886 acceptance/chaos_test.go:98  client 1 shutting down
I1215 02:44:57.426126 17886 acceptance/chaos_test.go:201  635 49.0/sec
I1215 02:44:57.579907 17886 acceptance/cluster/localcluster.go:523  stopping
--- FAIL: TestChaos (30.63s)
	chaos_test.go:167: failed to send RPC request Server.Batch: client is unhealthy
=== RUN   TestGossipPeerings
I1215 02:45:02.160923 17886 acceptance/cluster/localcluster.go:211  started docker-spy: 172.17.0.17
I1215 02:45:02.160955 17886 acceptance/cluster/localcluster.go:218  initializing cluster
I1215 02:45:05.212672 17886 acceptance/cluster/localcluster.go:326  creating ca (1024bit) in: /home/ubuntu/cockroach/.localcluster.certs.672848760
I1215 02:45:05.270569 17886 acceptance/cluster/localcluster.go:331  creating node (1024bit) certs in: /home/ubuntu/cockroach/.localcluster.certs.672848760
I1215 02:45:07.395121 17886 acceptance/cluster/localcluster.go:379  *** started roach0.local ***
  ui:    https://0.0.0.0:32777
  trace: https://0.0.0.0:32777/debug/requests
  logs:  /tmp/circle-artifacts.OpjNpm4/acceptance/roach0.local/cockroach.INFO
--
--- SKIP: TestBuildCluster (0.00s)
	terraform_test.go:33: only enabled during testing
=== RUN   TestFiveNodesAndWriters
--- SKIP: TestFiveNodesAndWriters (0.00s)
	main_test.go:54: running in docker mode
FAIL
ok github.com/cockroachdb/cockroach/acceptance 1337s

```
Please assign, take a look and update the issue accordingly.

--- a/acceptance/build_info_test.go
+++ b/acceptance/build_info_test.go
@@ -35,6 +35,8 @@ func TestBuildInfo(t *testing.T) {
 	l.Start()
 	defer l.AssertAndStop(t)
 
+	checkGossip(t, l, 20*time.Second, hasPeers(l.NumNodes()))
+
 	util.SucceedsWithin(t, 10*time.Second, func() error {
 		select {
 		case <-stopper:

--- a/acceptance/chaos_test.go
+++ b/acceptance/chaos_test.go
@@ -73,11 +73,10 @@ func TestChaos(t *testing.T) {
 		go func(i int) {
 			r, _ := randutil.NewPseudoRand()
 			value := randutil.RandBytes(r, 8192)
+			k := atomic.LoadInt64(&count)
 
 			for time.Now().Before(deadline) {
 				clients[i].RLock()
-				k := atomic.AddInt64(&count, 1)
-				atomic.AddInt64(&counts[i], 1)
 				v := value[:r.Intn(len(value))]
 				// TODO(bram): fix the retry options so the puts don't block
 				// occasionally for a full min during shutdown.
@@ -85,7 +84,7 @@ func TestChaos(t *testing.T) {
 					// These originate from DistSender when, for example, the
 					// leader is down. With more realistic retry options, we
 					// should probably not see them.
-					if _, ok := err.(*roachpb.SendError); ok || testutils.IsError(err, rpc.ErrShutdown.Error()) {
+					if _, ok := err.(*roachpb.SendError); ok || testutils.IsError(err, rpc.ErrShutdown.Error()) || testutils.IsError(err, "client is unhealthy") {
 						log.Warning(err)
 					} else {
 						errs <- err
@@ -93,6 +92,8 @@ func TestChaos(t *testing.T) {
 						return
 					}
 				}
+				k = atomic.AddInt64(&count, 1)
+				atomic.AddInt64(&counts[i], 1)
 				clients[i].RUnlock()
 			}
 			log.Infof("client %d shutting down", i)

